### DISCUSSION TITLE
How to set keymap in buffer of nvim-dap?

### DISCUSSION THREAD

**Discussion by NEX-S** [mfussenegger/nvim-dap#716] [Q&A]
I want to map j to be stepover like this:
```
vim.keymap.set('n', '<C-d>', dap.continue,  { noremap = true, silent = true, buffer = true })
vim.keymap.set('n', 'o', dap.step_out,  { noremap = true, silent = true, buffer = true })
vim.keymap.set('n', 'i', dap.step_into, { noremap = true, silent = true, buffer = true })
vim.keymap.set('n', 's', dap.terminate, { noremap = true, silent = true, buffer = true })
vim.keymap.set('n', 'j', dap.step_over, { noremap = true, silent = true, buffer = true })
vim.keymap.set('n', ';;'   , dap.toggle_breakpoint, { noremap = true, silent = true, buffer = true })
```

but after the nvim-dap exit, my normal keymap is not working.
So I'm wondering how to set keymap that only affects nvim-dap

**Comment by mfussenegger** [comment]
You can setup keymaps in a event listener when a debug session starts and remove them again if the debug session ends.

See https://github.com/mfussenegger/dotfiles/blob/e7abb9a13f8fb3075704ed703dd973ecf3502cc3/vim/.config/nvim/lua/me/dap.lua#L64-L75 for an example

**Reply by NEX-S** [reply]
It's fascinating! Appreciate for your reply!
